storage/pebble/registers.go

```diff
@@ -53,11 +64,12 @@ func (s *Registers) Get(
 	reg flow.RegisterID,
 	height uint64,
 ) (flow.RegisterValue, error) {
-	latestHeight := s.latestHeight.Load()
-	if height > latestHeight || height < s.firstHeight {
+	latestHeight := s.LatestHeight()
+	firstHeight := s.calculateFirstHeight(latestHeight)
+	if height > latestHeight || height < firstHeight {
 		return nil, errors.Wrap(
```

Comment:
Optimize for the happy case where height is above latestHeight, so we don't need to calculate first height.
```suggestion
	if height > latestHeight {
		return nil, fmt.Errorf(storage.ErrHeightNotIndexed, fmt.Sprintf("height %d not indexed, latestHeight: %d", height, latestHeight)
	} else {
		firstHeight := s.calculateFirstHeight(latestHeight)
		if height < firstHeight {
			return nil, fmt.Errorf("height %d not indexed, indexed range: [%d-%d], height, firstHeight, latestHeight)
		}			
	}
```

Reply:
nit
```golang
	if height > latestHeight {
		return nil, fmt.Errorf("height %d not indexed, latestHeight: %d, %w", height, latestHeight, storage.ErrHeightNotIndexed)
	}

	firstHeight := s.calculateFirstHeight(latestHeight)
	if height < firstHeight {
		return nil, fmt.Errorf("height %d not indexed, indexed range: [%d-%d], %w", height, firstHeight, latestHeight, storage.ErrHeightNotIndexed)
	}
```